mesonbuild/compilers.py

```diff
@@ -1761,7 +1761,7 @@ def get_warn_args(self, level):
         if level == '2':
             return ['-wi']
         else:
-            return ['-w']
+            return ['-wi']
```

Comment:
Yes, please don't merge this request yet, it might not actually solve the issue I am trying to fix and maybe this is a bug in LDC, even, since the documentation and the actual behavior differ...

So, I'll come back with an updated and better PR soonish.